test/common/testlib.py

```diff
@@ -1611,13 +1555,31 @@ def terminate_sessions():
         self.addCleanup(terminate_sessions)
 
     def tearDown(self):
+        success = self.checkSuccess()
```

Comment:
Is this a Pytest thing, it feels like a unittest thing. Actually, it's our own thing touching `unittest` internals. I admit that I am a bit confused how we managed this get rid of `_DebugOutcome` and thinking about it hmm if I `assert false` in the middle of a test function does that still mimick the same `sit()` behaviour. Either way this sounds interesting maybe for a separate PR?

Reply:
(Also wondering if pytest has something for this).

Reply:
A quick search gives some interesting approaches 

https://stackoverflow.com/questions/35703122/how-to-detect-when-pytest-test-case-failed

Reply:
https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures pytest specific

Reply:
My searching found https://stackoverflow.com/a/39606065, which is nicer than the one we have now.  It claims to work with pytest, but it wasn't working for me, so I just added another case to our existing one.  Maybe pytest also changed its internal API incompatibly.

I tested if this works properly with `set()` vs. cleanups and so on, and it does.  I'd be open to splitting it into a separate PR, since I agree that it's a pretty nice cleanup.